PlantnetBTP some identification response are greater than max

example:

```

2020-04-30T16:39:05.940463+00:00 app[web.1]: [31m[2020-04-30 16:39:05.940] [ERROR] [default] - [39mTwitterClient | POST statuses/update:{"in_reply_to_status_id":"1253682664839811078","status":"@JeanAlbertTurin - Bonjour, j'ai interrogé Pl@ntnet pour tenter d'identifier votre première image et voici en résultat : \nScore de 67.95% - nom scientifique : Sempervivum tectorum - famille: Crassulaceae - noms communs : Grande joubarbe, Joubarbe des toits, Artichaut bâtard\nAvec en illustration: fleur - Aguilar Belem / Pl@ntNet, cc-by-sa\nhttps://bs.floristic.org/v1/image/o/652f95fc08f4ab4d7c1fa793f75149b2c66ebb63\n\n#BetPlantnet,#IndentificationDePlantes"} - err:Error: Tweet needs to be a bit shorter.
2020-04-30T16:39:05.940736+00:00 app[web.1]: [31m[2020-04-30 16:39:05.940] [ERROR] [default] - [39m#BetPlantnet,#IndentificationDePlantes | Error: Tweet needs to be a bit shorter.
```

--- a/core/MyPlantnetClient.js
+++ b/core/MyPlantnetClient.js
@@ -3,6 +3,7 @@ var log4js = require('log4js');
 const fs = require('fs');
 const superagent = require('superagent');
 const queryString = require('querystring');
+const TinyURL = require('tinyurl');
 
 const MYPLANTNET_API_URL = 'https://my-api.plantnet.org/v2/identify/all';
 
@@ -32,13 +33,11 @@ class MyPlantnetClient {
 
   identify(imageUrl, doSimulate, cb) {
       this.logInfo("identify following image : " + imageUrl);
-      /*
       if (doSimulate) {
         let simulatedAnswer = fs.readFileSync('./core/data/plantNetFrenchResponse.json');
         cb(false, JSON.parse(simulatedAnswer));
         return;
       }
-      */
       // https://my.plantnet.org/account/doc // v2
       superagent.get(MYPLANTNET_API_URL)
       .query({
@@ -72,38 +71,50 @@ class MyPlantnetClient {
     let family =  aResult.species && aResult.species.family ? aResult.species.family.scientificNameWithoutAuthor : false;
     let commonNamesArray = aResult.species ? aResult.species.commonNames : false;
 
-    let infoOf = `Score de ${scorePercent}%`;
+    let infoOf = `(à ${scorePercent}%)`;
     if (scientificName) {
-      infoOf += ` - nom scientifique : ${scientificName}`;
+      infoOf += ` ${scientificName}`;
     }
     if (family) {
-      infoOf += ` - famille: ${family}`;
+      infoOf += ` (fam ${family})`;
     }
     if (this.arrayWithContent(commonNamesArray)) {
         let commonNamesArrayStr = commonNamesArray.join(', ');
-        infoOf += ` - noms communs : ${commonNamesArrayStr}`;
+        infoOf += ` com. ${commonNamesArrayStr}`;
     }
     return infoOf;
   }
 
-  resultImageOf(aResult) {
+  resultImageOf(aResult, cb) {
     if (!aResult) {
-      return false;
+      cb(false);
+      return;
     }
     let firstImage = aResult.images && aResult.images[0] ? aResult.images[0] : false;
     if (!firstImage) {
-      return false;
+      cb(false);
+      return;
     }
+    let pnClient = this;
     let firstImageUrl = firstImage.url ? firstImage.url : {};
     let imageUrl = firstImageUrl.o ? firstImageUrl.o : firstImageUrl.m ? firstImageUrl.m : firstImageUrl.s;
-    let imageCredits = firstImage.citation ? firstImage.citation : firstImage.author;
+    TinyURL.shorten(imageUrl, function(res, err) {
+        if (err) {
+            cb(pnClient.resultImageOfMessage(firstImage, imageUrl));
+        }
+        cb(pnClient.resultImageOfMessage(firstImage, res));
+    });
+  }
+
+  resultImageOfMessage(firstImage, shortenUrl) {
+    let imageCredits = firstImage.author; // firstImage.citation is too long for a tweet constraint
     let imageOrgan = firstImage.organ === 'flower' ? "fleur" :
                      firstImage.organ === 'leaf' ? 'feuille' : false;
     let imageOf = imageCredits;
     if (imageOrgan) {
         imageOf = `${imageOrgan} - ${imageOf}`;
     }
-    return `${imageOf}\n${imageUrl}`;
+    return `${imageOf}\n${shortenUrl}`;
   }
 
   hasScoredResult(plantnetResponse, minimalScore) {

--- a/core/TwitterClient.js
+++ b/core/TwitterClient.js
@@ -1,7 +1,9 @@
 /*jshint esversion: 6 */
 const dateFormat = require('dateformat');
 const Twit = require('twit');
-var log4js = require('log4js');
+const log4js = require('log4js');
+
+const TWEET_MAX_LENGTH=280;
 
 class TwitterClient {
   constructor() {
@@ -71,14 +73,20 @@ class TwitterClient {
   }
 
   replyTo(tweet, replyMsg, doSimulate, cb) {
-      let replyStatus = '@' + tweet.user.screen_name + ' - ' + replyMsg;
+      let replyStatus = `@${tweet.user.screen_name} ${replyMsg}`;
+      let replyLength = replyStatus.length;
+      if (replyLength > TWEET_MAX_LENGTH) {
+        let cause = `(${replyStatus.length} > ${TWEET_MAX_LENGTH})`;
+        cb(`replyStatus too long ${cause} : ${replyStatus}`);
+        return;
+      }
       let params = {
         in_reply_to_status_id: tweet.id_str,
         status: replyStatus
       };
 
       this.logDebug("Uses params to reply to : " + JSON.stringify(params));
-      this.logInfo("Plan to reply to => " + this.tweetLinkOf(tweet));
+      this.logInfo(`Plan to reply(${replyLength}) to => ` + this.tweetLinkOf(tweet));
       this.logInfo(" " + this.tweetInfoOf(tweet));
 
       if (doSimulate) {

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "botEnTrain",
-  "version": "1.1.0",
+  "version": "1.1.1",
   "description": "Twitter botEnTrain",
   "main": "bet.js",
   "scripts": {
@@ -29,6 +29,7 @@
     "fs": "0.0.1-security",
     "log4js": "^0.6.28",
     "superagent": "^5.2.2",
+    "tinyurl": "^1.1.5",
     "twit": "^2.2.11"
   }
 }

--- a/plugins/PlantnetBTP.js
+++ b/plugins/PlantnetBTP.js
@@ -3,6 +3,7 @@ const log4js = require('log4js');
 
 const PLANTNET_MINIMAL_PERCENT = 60;
 const PLANTNET_MINIMAL_RATIO = PLANTNET_MINIMAL_PERCENT / 100;
+const PLANTNET_SIMULATE = false;
 
 class PlantnetBTP {
   constructor(twitterClient, plantnetClient) {
@@ -69,7 +70,7 @@ class PlantnetBTP {
         }
         this.logDebug("candidateImage : " + candidateImage);
 
-        this.plantnetClient.identify(candidateImage, doSimulate, (err, plantResult) => {
+        this.plantnetClient.identify(candidateImage, PLANTNET_SIMULATE, (err, plantResult) => {
             if (err) {
                 this.logError(err);
                 cb({"message": "impossible d'identifier l'image", "status": 500});
@@ -92,11 +93,10 @@ class PlantnetBTP {
   }
 
   replyScoredResult(doSimulate, tweetCandidate, firstScoredResult, cb) {
-      let illustrateImage = this.plantnetClient.resultImageOf(firstScoredResult);
-      let replyMessage = "Bonjour, j'ai interrogé Pl@ntnet pour tenter d'identifier votre première image" +
-      " et voici en résultat : \n" +
+    this.plantnetClient.resultImageOf(firstScoredResult, (illustrateImage) => {
+      let replyMessage = "Pl@ntnet identifie " +
       this.plantnetClient.resultInfoOf(firstScoredResult) + "\n" +
-      (illustrateImage ? "Avec en illustration: " + illustrateImage : "");
+      (illustrateImage ? "\n\n" + illustrateImage : "");
 
       this.replyTweet(doSimulate, tweetCandidate, replyMessage, (err, replyTweet) => {
           if (err) {
@@ -117,6 +117,7 @@ class PlantnetBTP {
                   this.twitterClient.tweetInfoOf(replyTweet) + "\n"
           });
       });
+    });
   }
 
   replyNoScoredResult(doSimulate, tweetCandidate, cb) {

